src/corelibs/U2View/src/ov_msa/exclude_list/MsaExcludeList.h

```diff
@@ -0,0 +1,201 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#ifndef _U2_MSA_EXCLUDE_LIST_H_
+#define _U2_MSA_EXCLUDE_LIST_H_
+
+#include <QSet>
+#include <QWidget>
+
+#include <U2Core/DNASequence.h>
+#include <U2Core/global.h>
+
+#include <U2Gui/ObjectViewModel.h>
+
+class QLabel;
+class QListWidget;
+class QListWidgetItem;
+class QPlainTextEdit;
+class QSplitter;
+class QToolButton;
+
+namespace U2 {
+
+class LoadDocumentTask;
+class MSAEditor;
+class MaModificationInfo;
+class MsaExcludeListWidget;
+class MultipleAlignment;
+class MultipleAlignmentRow;
+
+/** MSA editor built-in support for "MSA Exclude List" widget. */
+class MsaExcludeListContext : public GObjectViewWindowContext {
+    Q_OBJECT
+public:
+    MsaExcludeListContext(QObject *parent);
+
+    QAction *getMoveMsaSelectionToExcludeListAction(MSAEditor *msaEditor);
+
+protected:
+    void initViewContext(GObjectView *view) override;
+
+private:
+    void toggleExcludeListView(MSAEditor *msaEditor);
+
+    void moveMaSelectionToExcludeList(MSAEditor *msaEditor);
+
+    void updateState(MSAEditor *msaEditor);
+
+    MsaExcludeListWidget *findActiveExcludeList(MSAEditor *msaEditor);
+
+    MsaExcludeListWidget *openExcludeList(MSAEditor *msaEditor);
+
+    void updateMsaEditorSplitterStyle(MSAEditor *msaEditor);
+};
+
+/** Stores per step (Undo or Redo) information used by Exclude List to keep synchronized state with MSA Editor during Undo/Redo ops. */
+struct UndoRedoStep {
+    /** Direction of the original move. */
+    bool isMoveFromMsaToExcludeList = false;
+    /** Affected Exclude List row ids. The row ids keeps the same order as MSA rows being changed during the move. */
+    QList<int> excludeListRowIdsDelta;
+};
+
+/**
+ * Exclude list widget component for MsaEditor.
+ * The 'exclude list' renders a list of sequences excluded from the multiple alignment object.
+ *
+ * The widget contains a data model for Exclude List and a limited Undo/Redo support for MSA:
+ * the Undo-Redo is supported only for the life-time of the widget: when Exclude List is closed the Undo/Redo stack of the widget is reset.
+ */
+class MsaExcludeListWidget : public QWidget {
+    Q_OBJECT
+public:
+    MsaExcludeListWidget(QWidget *parent, MSAEditor *msaEditor, MsaExcludeListContext *viewContext);
+
+    /** Triggers auto-save of the current exclude list state. */
+    ~MsaExcludeListWidget();
```

Comment:
As it was already told, the destructor shouldn't trigger anything, it should just deallocate memory

Reply:
Please read the comment above. Destructor is a tool and there is nothing wrong with it when it is used correctly.
BTW You ask me to use a signal that emitted from the destructor itself.